Assets/Scripts/Models/RoomManager.cs

```diff
@@ -0,0 +1,320 @@
+﻿using System;
+using System.Collections.Generic;
+
+public class RoomManager
+{
+    public event Action<Room> Adding;
+
+    public event Action<Room> Added;
+
+    public event Action<Room, Room> Split;
+
+    public event Action<Room, Room> Joined;
+
+    public event Action<Room> Removing;
+
+    public event Action<Room> Removed;
+
+    public List<Room> rooms;
```

Comment:
I could try to do something that yields every room on the list or something like that... it might work.. Stay tuned!